src/scipp/core/variable.py

```diff
@@ -782,7 +792,23 @@ def arange(dim: str,
                                              start=start,
                                              stop=stop,
                                              step=step)
-    return array(dims=[dim], values=_np.arange(**range_args), unit=unit, dtype=dtype)
+    args = [x for x in [start, stop, step] if x is not None]
+    types = [
+        x.values.dtype if isinstance(x, _cpp.Variable) else _np.asarray(x).dtype
+        for x in args
+    ]
+    if dtype is None:
+        candidates = set(types)
+        if len(candidates) == 1:
+            dtype = next(iter(candidates))
+    if dtype is not None and dtype != str('datetime64'):
+        numpy_dtype = str(dtype) if isinstance(dtype, DType) else dtype
```

Comment:
Is this `if` required?

Reply:
Yes, because NumPy cannot handle `scipp.DType`.